### PR TITLE
Migrate from OpenAI Gym to Farama Foundation Gymnasium

### PR DESCRIPTION
**Pull request by StephenOman**
**Breaking Changes**

Migration from OpenAI Gym to the Farama Foundation Gymnasium introduces breaking modifications to some API calls.

For details, see [v21 to v26 Migration Guide](https://gymnasium.farama.org/content/migration-guide/) on the Farama Foundation website.

Main changes:
- Seeding can only take place on reset()
- reset() also allows options to be passed into the environment
- reset() returns additional information
- step() returns an additional value to indicate if the episode was truncated as the fourth item in the tuple
- render() no longer accepts the rendering mode as a parameter. Instead, rendering mode should be initialised when the environment is created in gym.make()

This PR also includes documentation changes and some minor bug fixes.

### CODE REVIEW

**Comment by StephenOman** [comment]
When this is released, a small change is needed to `test_package.yml:31`:

`python -c 'import nle; import gymnasium as gym; e = gym.make("NetHack-v0"); e.reset(); e.step(0)'`

File: doc/nle/source/getting_started.rst
Comment (heiner) [inline doc/nle/source/getting_started.rst:7]: Not the fault of this PR but it says `cmake>=3.15` above. Let's perhaps just drop the version specification in this file?

File: doc/nle/source/getting_started.rst
Comment (heiner) [inline doc/nle/source/getting_started.rst:66]: Is `1` intuitive here? Or should we instead do a variable lookup?

File: nle/tests/test_db.py
Comment (heiner) [inline nle/tests/test_db.py:29]: Just out of curiosity, what makes the `unwrapped` necessary here and below?

**Review by heiner — APPROVED** [review]
Thanks! I have not tested this but the changes LGTM.

Once this landed and we updated the other test, we should also bump the version (to `0.9.3`, or if you're daring to `1.0`?) and then do a release to PyPI.

**Comment by StephenOman** [comment]
> Once this landed and we updated the other test, we should also bump the version (to `0.9.3`, or if you're daring to `1.0`?) and then do a release to PyPI.

I definitely vote for a major version change since there are breaking changes in the API. Given the change to new repo etc., `1.0.0` is perfect.

File: doc/nle/source/getting_started.rst
Comment (StephenOman) [reply doc/nle/source/getting_started.rst]: I'd rather have the documentation be consistent with the underlying specification. Right now, `pyproject.toml` says `cmake>=3.18` so I've updated both `README.md` and `getting_started.rst` to match (new commit).

File: doc/nle/source/getting_started.rst
Comment (StephenOman) [reply doc/nle/source/getting_started.rst:66]: Those lines are a direct copy of the code in README.md. 

It is a good suggestion. In the test I added for truncation, it uses this step formulation:

`obs, reward, termination, truncation, info = env.step(nethack.MiscDirection.WAIT)`

rather than the ordinal number of the WAIT action. But it would be nice to have something on the lines of:

```
from nethack import actions
...
env.step(actions['WAIT'])
```

Sounds like a Request for Enhancement :)

File: nle/tests/test_db.py
Comment (StephenOman) [reply nle/tests/test_db.py:29]: The Gymnasium API deprecated the original way of accessing wrapped attributes with the intention of removing it completely in the upcoming v1.0. Using `unwrapped` is their suggestion to properly access these.

File: doc/nle/source/getting_started.rst
Comment (heiner) [reply doc/nle/source/getting_started.rst]: If you prefer that, fine. I feel it's better to keep the number of places in the docs that need to be updated for this small, hence to remove it from one location. Otherwise things more easily get out of sync as they did here.

File: doc/nle/source/getting_started.rst
Comment (heiner) [reply doc/nle/source/getting_started.rst:66]: Agreed we should have one enum with all actions. Kinda silly we don't, actually. One added complexity is that the RL actions are `0` to `N-1` while the underlying NetHack actions are keyboard commands where e.g. `WAIT` is space. (Another problem is that actions change their meaning given the context, e.g. for itemized lists.)

**Comment by heiner** [comment]
1.0.0 sgtm.

**Pull request merged**
